### PR TITLE
[breaking change] Expose fluentd config without ENV vars - Take 2

### PR DESCRIPTION
**Pull request by samjsong**
###### Description

The goal was to expose the Fluentd config to the user, while hopefully also making the `values.yaml` easier to use. We also want to get rid of the layer of indirection that was created with the ENV vars.

`values.yaml` will be the most interesting file to look at, hopefully the rest of it is just inserting the mustache syntax in the correct places to reflect what is in `values.yaml`.

Note that we also removed several configs in the process, including `addStream`, `addTime`, and `addTimestamp`, which if desired could be done through use of a `record_modifier` plugin. The relevant code was also removed from the `kubernetes_sumologic` filter plugin code.

Reading commit by commit might be easier, I tried to break it up into reasonable chunks but sometimes I made mistakes or missed things that I fixed in later commits so it's not perfect... sorry 😞 

###### Testing performed

- [x] ci/build.sh
- [x] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by samjsong** [comment]
CI failed due to manual changes in `fluentd-sumologic.yaml.tmpl`, I think there might have been manual changes when I merged master at some point, but I thought that by pulling master for that file that would be fixed... I'll take a look at that later

File: deploy/helm/sumologic/conf/logs/logs.source.containers.conf [hunk 1]
```diff
@@ -18,45 +18,42 @@
...
-    exclude_container_regex "#{ENV['EXCLUDE_CONTAINER_REGEX']}"
-    exclude_host_regex "#{ENV['EXCLUDE_HOST_REGEX']}"
+    source_name {{ .Values.fluentd.logs.containers.sourceName | quote }}
+    source_host {{ .Values.fluentd.logs.containers.sourceHost | quote }}
+    log_format {{ .Values.fluentd.logs.output.logFormat | quote }}
+    source_category {{ .Values.fluentd.logs.containers.sourceCategory | quote }}
+    source_category_prefix {{ .Values.fluentd.logs.containers.sourceCategoryPrefix | quote }}
+    source_category_replace_dash {{ .Values.fluentd.logs.containers.sourceCategoryReplaceDash | quote }}
+    exclude_namespace_regex {{ .Values.fluentd.logs.containers.excludeNamespaceRegex | quote }}
+    exclude_pod_regex {{ .Values.fluentd.logs.containers.exclud8ePodRegex | quote }}
```
Comment (maimaisie) [inline deploy/helm/sumologic/conf/logs/logs.source.containers.conf]: typo in `exclud8ePodRegex`

File: deploy/kubernetes/fluentd-sumologic.yaml.tmpl [hunk 2]
```diff
@@ -15,10 +15,11 @@ data:
     @include logs.conf
   buffer.output.conf: |-
     compress gzip
-    flush_interval "#{ENV['FLUSH_INTERVAL']}"
-    flush_thread_count "#{ENV['NUM_THREADS']}"
-    chunk_limit_size "#{ENV['CHUNK_LIMIT_SIZE']}"
-    total_limit_size "#{ENV['TOTAL_LIMIT_SIZE']}"
+    flush_interval "5s"
+    flush_thread_count "4"
```
Comment (vsinghal13) [inline deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: Should these contain double quotes for numeric values as well?

File: deploy/kubernetes/fluentd-sumologic.yaml.tmpl [hunk 3]
```diff
@@ -217,23 +220,24 @@ data:
...
       <filter host.kubelet.**>
         @type kubernetes_sumologic
-        source_category kubelet
-        source_name k8s_kubelet
-        source_category_prefix "#{ENV['SOURCE_CATEGORY_PREFIX']}"
-        exclude_facility_regex "#{ENV['EXCLUDE_FACILITY_REGEX']}"
-        exclude_host_regex "#{ENV['EXCLUDE_HOST_REGEX']}"
-        exclude_priority_regex "#{ENV['EXCLUDE_PRIORITY_REGEX']}"
-        exclude_unit_regex "#{ENV['EXCLUDE_UNIT_REGEX']}"
+        source_category "kubelet"
```
Comment (vsinghal13) [inline deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: there were no quotes before for this.

File: deploy/helm/sumologic/conf/logs/logs.source.containers.conf (hunk 1 above)
Comment (samjsong) [reply deploy/helm/sumologic/conf/logs/logs.source.containers.conf]: great catch, thanks Maisie

File: deploy/kubernetes/fluentd-sumologic.yaml.tmpl (hunk 2 above)
Comment (samjsong) [reply deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: I believe quotes are fine for these values, Fluentd parses these parameters in the correct format, based on https://docs.fluentd.org/configuration/config-file#supported-data-types-for-values

File: deploy/kubernetes/fluentd-sumologic.yaml.tmpl (hunk 3 above)
Comment (samjsong) [reply deploy/kubernetes/fluentd-sumologic.yaml.tmpl]: These values are parsed as string, and the quotes are removed when taking the value into account

**Comment by samjsong** [comment]
Thank you guys for reviewing! before we merge I'd like to write up the list of changes and maybe some documentation on how you can preserve certain functionality with the `record_modifier` plugin.

**Comment by samjsong** [comment]
Closing in favor of #428 due to merge conflicts

**Pull request closed without merging**